content/200-concepts/150-data-platform/08-data-proxy.mdx

```diff
@@ -48,125 +48,162 @@ Connection strings always have the following format:
+
+### Step 4: Generate the client
+
+To generate Prisma Client with the Data Proxy, call `prisma generate` with the option `--data-proxy`:
 
 ```terminal
-PRISMA_CLIENT_ENGINE_TYPE='dataproxy' npx prisma generate
+prisma generate --data-proxy
 ```
 
-#### Step 6: Start your app
+When you use `--data-proxy`, you get a very lightweight Prisma Client runtime of approximately 250kb. This is because the Data Proxy does not need local engine files, so the generation process does not include the [query engine](/concepts/components/prisma-engines/query-engine).
 
-If you are loading the connection string in your Prisma Schema using `env()`, there should be no further changes needed other than verifying that the environment variable changed to your Data Proxy connection string. For example:
+### Step 5: Import and instantiate Prisma Client in your code
 
-```terminal
-DATABASE_URL='prisma://aws-us-east-1.prisma-data.com/?api_key=...' npm start
+#### Node.js
+
+To use the Data Proxy with Node.js, use `@prisma/client` as described in [Generating the client](/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client).
+
+#### Edge runtimes
+
+To use the Data Proxy with Cloudflare Workers or Vercel Edge Functions, you need to import `@prisma/client/edge` instead of the usual `@prisma/client`, as follows:
+
+ <TabbedContent tabs={[<FileWithIcon text="TypeScript" icon="code"/>, <FileWithIcon text="JavaScript" icon="code"/>]}>
+
+ <tab>
+
+```ts
+import { PrismaClient } from '@prisma/client/edge'
+const prisma = new PrismaClient()
+// use `prisma` in your application to read and write data in your DB
 ```
 
-For guidance on how to configure environment variables on your deployment, please refer to your provider.
+  </tab>
 
-- AWS: [Using AWS Lambda environment variables](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html)
-- Heroku: [Configuration and Config Vars](https://devcenter.heroku.com/articles/config-vars)
+  <tab>
+
+```js
+const { PrismaClient } = require('@prisma/client/edge')
+const prisma = new PrismaClient()
+// use `prisma` in your application to read and write data in your DB
+```
 
-### Error codes
+</tab>
 
-Because Data Proxy connections operate differently from a local database connection, error codes will be different from regular ones. In general, all Data Proxy-related errors will start with `P5xxx`.
+</TabbedContent>
 
 <Admonition type="warning">
 
-**Subject to change**<br />
-Please note that the Prisma Data Proxy public interface is not stable yet. The listed codes may be changed at any time without notice.
+You can use `@prisma/client/edge` only with the Data Proxy. Make sure to use the `--data-proxy` flag when you [generate the client](#step-4-generate-the-client).
 
 </Admonition>
 
-#### P5000
+### Step 6: Start your app
+
+You are now ready to try the Data Proxy. Go ahead and start your app!
+
+## Error codes
```

Comment:
I think the **Error codes** should come after **Important considerations about the Data Proxy**.

Or better yet, the Error codes could live on a separate page at some point – similar to https://www.prisma.io/docs/reference/api-reference/error-reference#error-codes

Reply:
Fair point - I'll move the error codes section downwards.

Reply:
We're going to do a docs re-org in the near future, and having a separate page is something we'll consider.